src/main/java/io/orkes/conductor/client/automator/EnvironmentVariableReader.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2022 Orkes, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package io.orkes.conductor.client.automator;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Reads and processes environmental variables related to Conductor worker
+ * configuration. The following environmental variable pattern is supported:
+ * - CONDUCTOR_WORKER_.*_DOMAIN
+ *
+ * The pattern allows users to define custom task-specific configuration by
+ * setting environmental variables with a specific format. The "TASK_NAME" part
+ * in the pattern represents the name of the task for which the configuration is
+ * intended. The actual task name can replace the ".*" portion in the pattern.
+ *
+ * Note: To ensure compatibility with all operating systems, it is recommended
+ * to use underscores (_) instead of dots (.) within the task name.
+ *
+ * Examples:
+ * - CONDUCTOR_WORKER_EMAIL_NOTIFICATION_DOMAIN: Configuration for the
+ * "EMAIL_NOTIFICATION" task.
+ * - CONDUCTOR_WORKER_Em@il_N0T1F1C4TI0N_DOMAIN: Configuration for the
+ * "Em@il_N0T1F1C4TI0N" task.
+ * - CONDUCTOR_WORKER_DATA_PROCESSING_DOMAIN: Configuration for the
```

Comment:
minor nit: workers names are rarely all upper case and this will confuse user if they need to upper case the worker name.